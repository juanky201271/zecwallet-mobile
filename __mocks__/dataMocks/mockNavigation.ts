import { StackScreenProps } from '@react-navigation/stack';

export const mockNavigation: StackScreenProps<any>['navigation'] = {
  // Propiedades y métodos necesarios para la navegación
  navigate: jest.fn(),
  goBack: jest.fn(),
  dispatch: jest.fn(),
  reset: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getId: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  // Agrega cualquier otra propiedad o método necesario para tu caso
};
