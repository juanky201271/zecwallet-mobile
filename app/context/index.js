import { defaultAppContextLoaded, ContextAppLoaded, ContextAppLoadedProvider } from './contextAppLoaded';
import { defaultAppContextLoading, ContextAppLoading, ContextAppLoadingProvider } from './contextAppLoading';

export {
  defaultAppContextLoaded,
  ContextAppLoaded,
  ContextAppLoadedProvider,
  defaultAppContextLoading,
  ContextAppLoading,
  ContextAppLoadingProvider,
};
