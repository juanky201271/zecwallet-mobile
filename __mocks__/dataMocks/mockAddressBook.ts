import { AddressBookFileClass } from '../../app/AppState';

export const mockAddressBook: AddressBookFileClass[] = [
  {
    label: 'pepe',
    address: 'u1234567890_____________',
  },
  {
    label: 'lolo',
    address: 'u0987654321_____________',
  },
];
