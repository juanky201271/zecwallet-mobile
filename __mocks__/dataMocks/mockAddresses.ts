import { AddressClass, AddressKindEnum, ReceiverEnum } from '../../app/AppState';

export const mockAddresses: AddressClass[] = [
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'UA-12345678901234567890',
    addressKind: AddressKindEnum.u,
    containsPending: false,
    receivers: ReceiverEnum.o + ReceiverEnum.z + ReceiverEnum.t,
  },
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'sapling-12345678901234567890',
    addressKind: AddressKindEnum.z,
    containsPending: false,
    receivers: ReceiverEnum.z,
  },
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'transparent-12345678901234567890',
    addressKind: AddressKindEnum.t,
    containsPending: false,
    receivers: ReceiverEnum.t,
  },
];
