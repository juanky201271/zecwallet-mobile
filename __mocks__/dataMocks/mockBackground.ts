import { BackgroundType } from '../../app/AppState';

export const mockBackground: BackgroundType = {
  batches: 0,
  message: '',
  date: 0,
  dateEnd: 0,
};
