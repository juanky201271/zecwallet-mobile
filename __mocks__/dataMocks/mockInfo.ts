import { ChainNameEnum, CurrencyNameEnum, InfoType } from '../../app/AppState';
import { serverUris } from '../../app/uris';

export const mockInfo: InfoType = {
  chainName: ChainNameEnum.mainChainName,
  serverUri: serverUris(() => {})[0].uri,
  latestBlock: 2000000,
  connections: 0,
  version: 'server_version',
  verificationProgress: 0,
  currencyName: CurrencyNameEnum.ZEC,
  solps: 0,
  zingolib: 'zingolib_version',
};
