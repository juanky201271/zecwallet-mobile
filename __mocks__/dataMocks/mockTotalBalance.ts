import { TotalBalanceClass } from '../../app/AppState';

export const mockTotalBalance: TotalBalanceClass = {
  transparentBal: 0.12345678,
  privateBal: 0.4,
  spendablePrivate: 0.2,
  orchardBal: 0.6,
  spendableOrchard: 0.3,
  total: 1.12345678,
};
