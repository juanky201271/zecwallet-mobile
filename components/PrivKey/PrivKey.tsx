/* eslint-disable react-native/no-inline-styles */
import React, { useContext, useEffect, useState } from 'react';
import { View, ScrollView, SafeAreaView, TouchableOpacity, Text } from 'react-native';
import { useTheme } from '@react-navigation/native';
import Clipboard from '@react-native-community/clipboard';
import QRCode from 'react-native-qrcode-svg';

import FadeText from '../Components/FadeText';
import Utils from '../../app/utils';
import { ThemeType } from '../../app/types';
import { ContextAppLoaded } from '../../app/context';
import Header from '../Header';
import moment from 'moment';
import 'moment/locale/es';
import 'moment/locale/pt';
import 'moment/locale/ru';
import { SnackbarDurationEnum } from '../../app/AppState';

type PrivKeyProps = {
  closeModal: () => void;
  address: string;
  keyType: number;
  privKey: string;
};
const PrivKey: React.FunctionComponent<PrivKeyProps> = ({ address, keyType, privKey, closeModal }) => {
  const context = useContext(ContextAppLoaded);
  const { translate, addLastSnackbar, language } = context;
  const { colors } = useTheme() as unknown as ThemeType;
  moment.locale(language);

  const [expandAddress, setExpandAddress] = useState<boolean>(false);
  const [keyTypeString, setKeyTypeString] = useState<string>('');
  const [keyChunks, setKeyChunks] = useState<string[]>([]);

  useEffect(() => {
    const keyTypeStr = keyType === 0 ? translate('privkey.privkey') : translate('privkey.viewkey');

    // 30 characters per line
    const numLines = privKey.length < 40 ? 2 : privKey.length / 30;
    const keyChu = Utils.splitStringIntoChunks(privKey, Number(numLines.toFixed(0)));

    setKeyTypeString(keyTypeStr as string);
    setKeyChunks(keyChu);
  }, [keyType, privKey, translate]);

  const doCopy = () => {
    //if (address) {
    Clipboard.setString(privKey);
    addLastSnackbar({
      message: translate('privkey.tapcopy-message') as string,
      duration: SnackbarDurationEnum.short,
    });
    //}
  };

  return (
    <SafeAreaView
      style={{
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'stretch',
        height: '100%',
        backgroundColor: colors.background,
      }}>
      <Header
        title={keyTypeString + ' ' + translate('privkey.title')}
        noBalance={true}
        noSyncingStatus={true}
        noDrawMenu={true}
        noPrivacy={true}
        closeScreen={closeModal}
      />
      <ScrollView
        style={{ maxHeight: '90%' }}
        contentContainerStyle={{
          flexDirection: 'column',
          alignItems: 'stretch',
          justifyContent: 'flex-start',
        }}>
        <View
          style={{ display: 'flex', flexDirection: 'column', marginTop: 0, alignItems: 'center', marginBottom: 30 }}>
          <View style={{ alignItems: 'center', paddingBottom: 0, paddingTop: 10 }}>
            <FadeText style={{ color: colors.text, textAlign: 'center', marginLeft: 10, marginRight: 10 }}>
              {translate('privkey.address') as string}
            </FadeText>
            <TouchableOpacity
              onPress={() => {
                setExpandAddress(true);
              }}>
              <FadeText style={{ textAlign: 'center', marginLeft: 10, marginRight: 10 }}>
                {expandAddress ? address : Utils.trimToSmall(address, 10)}
              </FadeText>
            </TouchableOpacity>
          </View>

          <View style={{ padding: 10, backgroundColor: colors.border, marginTop: 15, marginBottom: 20 }}>
            <QRCode
              value={privKey}
              size={225}
              ecl="L"
              backgroundColor={colors.border}
              logo={require('../../assets/img/logobig-zingo.png')}
              logoSize={35}
              logoBackgroundColor={colors.border}
              logoBorderRadius={10} /* android not soported */
              logoMargin={5}
            />
          </View>
          <TouchableOpacity onPress={doCopy}>
            <Text style={{ color: colors.text, textDecorationLine: 'underline', marginBottom: 5, minHeight: 48 }}>
              {translate('seed.tapcopy') as string}
            </Text>
          </TouchableOpacity>

          {keyChunks.map(c => (
            <FadeText
              key={c}
              style={{
                flexBasis: '100%',
                textAlign: 'center',
                fontFamily: 'verdana',
                fontSize: 18,
                color: colors.text,
              }}>
              {c}
            </FadeText>
          ))}
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

export default PrivKey;
