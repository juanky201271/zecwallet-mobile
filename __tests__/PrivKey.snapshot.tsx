/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import PrivKey from '../components/PrivKey';
import { defaultAppContextLoaded, ContextAppLoadedProvider } from '../app/context';
import { mockTranslate } from '../__mocks__/dataMocks/mockTranslate';
import { mockInfo } from '../__mocks__/dataMocks/mockInfo';
import { mockTotalBalance } from '../__mocks__/dataMocks/mockTotalBalance';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.',
      groupingSeparator: ',',
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-community/netinfo', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RNCNetInfo = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});

// test suite
describe('Component PrivKey - test', () => {
  //snapshot test
  const state = defaultAppContextLoaded;
  state.translate = mockTranslate;
  state.info = mockInfo;
  state.totalBalance = mockTotalBalance;
  const onClose = jest.fn();
  test('PrivKey Private - snapshot', () => {
    const privKey = render(
      <ContextAppLoadedProvider value={state}>
        <PrivKey
          address={'UA-12345678901234567890'}
          keyType={0}
          privKey={'priv-key-12345678901234567890'}
          closeModal={onClose}
        />
      </ContextAppLoadedProvider>,
    );
    expect(privKey.toJSON()).toMatchSnapshot();
  });
  test('PrivKey View - snapshot', () => {
    const privKey = render(
      <ContextAppLoadedProvider value={state}>
        <PrivKey
          address={'UA-12345678901234567890'}
          keyType={1}
          privKey={'view-key-12345678901234567890'}
          closeModal={onClose}
        />
      </ContextAppLoadedProvider>,
    );
    expect(privKey.toJSON()).toMatchSnapshot();
  });
});
