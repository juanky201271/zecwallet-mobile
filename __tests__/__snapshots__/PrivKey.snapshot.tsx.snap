// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PrivKey - test PrivKey Private - snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(255, 255, 255)",
          "display": "flex",
          "minHeight": 25,
          "paddingBottom": 0,
          "paddingTop": 10,
          "zIndex": -1,
        }
      }
      testID="header"
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "marginHorizontal": 5,
            "marginTop": 12,
          }
        }
      />
    </View>
    <View
      style={
        {
          "left": 0,
          "padding": 11.5,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 45,
          }
        }
      />
    </View>
    <View
      style={
        {
          "padding": 15,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/img/logobig-zingo.png",
          }
        }
        style={
          {
            "borderRadius": 10,
            "height": 38,
            "resizeMode": "contain",
            "width": 38,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <
          color="rgb(0, 122, 255)"
          icon={
            {
              "icon": [
                320,
                512,
                [
                  9001,
                ],
                "f053",
                "M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z",
              ],
              "iconName": "chevron-left",
              "prefix": "fas",
            }
          }
          size={30}
          style={
            {
              "marginHorizontal": 10,
            }
          }
        />
      </View>
      <Text
        selectable={true}
        style={
          {
            "color": "rgb(28, 28, 30)",
            "fontSize": 18,
            "fontWeight": "600",
            "opacity": 1,
            "paddingHorizontal": 5,
          }
        }
      >
        text translated text translated
      </Text>
      <View
        style={
          {
            "height": 30,
            "marginHorizontal": 10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(0, 122, 255)",
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    style={
      {
        "maxHeight": "90%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginBottom": 30,
            "marginTop": 0,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "paddingBottom": 0,
              "paddingTop": 10,
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(28, 28, 30)",
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 0.65,
                "textAlign": "center",
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "opacity": 0.65,
                  "textAlign": "center",
                }
              }
            >
              UA-1234567...1234567890
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "rgb(216, 216, 216)",
              "marginBottom": 20,
              "marginTop": 15,
              "padding": 10,
            }
          }
        >
          <mockQRCode />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "marginBottom": 5,
                "minHeight": 48,
                "textDecorationLine": "underline",
              }
            }
          >
            text translated
          </Text>
        </View>
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "flexBasis": "100%",
              "fontFamily": "verdana",
              "fontSize": 18,
              "opacity": 0.65,
              "textAlign": "center",
            }
          }
        >
          priv-key-123456
        </Text>
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "flexBasis": "100%",
              "fontFamily": "verdana",
              "fontSize": 18,
              "opacity": 0.65,
              "textAlign": "center",
            }
          }
        >
          78901234567890
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Component PrivKey - test PrivKey View - snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(255, 255, 255)",
          "display": "flex",
          "minHeight": 25,
          "paddingBottom": 0,
          "paddingTop": 10,
          "zIndex": -1,
        }
      }
      testID="header"
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "marginHorizontal": 5,
            "marginTop": 12,
          }
        }
      />
    </View>
    <View
      style={
        {
          "left": 0,
          "padding": 11.5,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 45,
          }
        }
      />
    </View>
    <View
      style={
        {
          "padding": 15,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/img/logobig-zingo.png",
          }
        }
        style={
          {
            "borderRadius": 10,
            "height": 38,
            "resizeMode": "contain",
            "width": 38,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <
          color="rgb(0, 122, 255)"
          icon={
            {
              "icon": [
                320,
                512,
                [
                  9001,
                ],
                "f053",
                "M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z",
              ],
              "iconName": "chevron-left",
              "prefix": "fas",
            }
          }
          size={30}
          style={
            {
              "marginHorizontal": 10,
            }
          }
        />
      </View>
      <Text
        selectable={true}
        style={
          {
            "color": "rgb(28, 28, 30)",
            "fontSize": 18,
            "fontWeight": "600",
            "opacity": 1,
            "paddingHorizontal": 5,
          }
        }
      >
        text translated text translated
      </Text>
      <View
        style={
          {
            "height": 30,
            "marginHorizontal": 10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(0, 122, 255)",
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    style={
      {
        "maxHeight": "90%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginBottom": 30,
            "marginTop": 0,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "paddingBottom": 0,
              "paddingTop": 10,
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(28, 28, 30)",
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 0.65,
                "textAlign": "center",
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "opacity": 0.65,
                  "textAlign": "center",
                }
              }
            >
              UA-1234567...1234567890
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "rgb(216, 216, 216)",
              "marginBottom": 20,
              "marginTop": 15,
              "padding": 10,
            }
          }
        >
          <mockQRCode />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "marginBottom": 5,
                "minHeight": 48,
                "textDecorationLine": "underline",
              }
            }
          >
            text translated
          </Text>
        </View>
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "flexBasis": "100%",
              "fontFamily": "verdana",
              "fontSize": 18,
              "opacity": 0.65,
              "textAlign": "center",
            }
          }
        >
          view-key-123456
        </Text>
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "flexBasis": "100%",
              "fontFamily": "verdana",
              "fontSize": 18,
              "opacity": 0.65,
              "textAlign": "center",
            }
          }
        >
          78901234567890
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
